docs/contributor/updateability.md

```diff
@@ -0,0 +1,349 @@
+
+`/var/lib/microshift-backups/` also stores backups of MicroShift data.
+Backups are created by copying data directory with `--reflink`
+option to leverage Copy-on-Write functionality.
+
+Naming schema of backup directories: `deployment-id_boot-id[_unhealthy]`.
+
+Suffix `_unhealthy` is used to mark backups of unhealthy systems.
+These backups are performed when all existing metadata suggests that it is expected
+that MicroShift should wipe the data and start from clean state
+(for example "FDO" scenario where unhealthy deployment leaves stale data).
+
+### `/var/lib/microshift/version`
+
+TODO
+
+Schema:
+```
+MAJOR.MINOR.PATCH
+```
+
+All of MAJOR, MINOR, and PATCH are unsigned integers.
+For example: `4.14.0`
+
+## Greenboot [ostree-only]
+
+- `/etc/greenboot/green.d/40_microshift_set_healthy.sh`
+- `/etc/greenboot/red.d/40_microshift_set_unhealthy.sh`
+- `/etc/greenboot/red.d/40_microshift_pre_rollback.sh`
+
+## (rpm-)ostree [ostree-only]
+
+TODO
+
+### `rpm-ostree` vs `ostree`
+
+TODO
+
+### Commits and deployments
+
+TODO
+
+<!--
+ostree     -> commits
+rpm-ostree -> deployments
+
+Deployment is a commit with a boot entry.
+-->
+
+## Useful `ostree` commands
+
+TODO
+
+## Useful `rpm-ostree` commands
+
+TODO
+
+## Grub [ostree]
+
+TODO
+- boot_counter and boot_success
+
+### grub-boot-success.timer
+
+## Implementation
+
+Upgradeability introduces following features that run in following order:
+- backup management,
+- version metadata management,
+- Kubernetes storage migration.
+
+These three areas are mostly separate, but backup management
+may inspect version metadata to handle some corner cases.
+Backup management and version metadata management run
+before MicroShift attempts to start all the components and cluster itself.
+Storage migration runs when cluster is up and running 
+(and in 10 minutes intervals afterwards) because it
+requires runtime components to correctly perform its task.
+
+### Automated backup management [ostree-only]
+
+It runs very close to the start of MicroShift's `run` command.
+
+It checks for existence of `/run/ostree-booted` file.
+If the file is not present, it is assumed that the system is regular RPM based
+and this stage is skipped without an error.
+
+#### Handling unusual scenarios
+
+Then, procedure checks existence of three on-disk artifacts:
+- `/var/lib/microshift` (ignores `.nodename`[0]) - referred to as `data` below
+- `/var/lib/microshift/version`
+- `/var/lib/microshift-backups/health.json`
+
+[0] `.nodename` might be created by the "effective config generation" phase
+which happens before the prerun. This can create `.nodename` and result
+in `/var/lib/microshift` being present but, beside that one file, empty.
+
+##### `/var/lib/microshift` (data) does not exist
+
+> Lack of `/var/lib/microshift` implies that `/var/lib/microshift/version` does not exist.
+
+If both `data` and `health.json` are missing, then we assume it's a first run of MicroShift.
```

Comment:
nit: Using "`data`" here makes me think that's the name of an actual file, but I think you mean `/var/lib/microshift` == `data`? Maybe use _data_ instead?